Update 1.8.18

zenhack opened the pull request:
This updates to version 1.8.18, and fixes conflicts in patches. Not working yet, so marking this one as a draft; on startup I get:

```
An error occurred while loading the page:

Uncaught Error: The module at "js-cookie/dist/js.cookie" does not exist.
in http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623
at line 559:9

Stack trace:

Error: The module at "js-cookie/dist/js.cookie" does not exist.
    at _designatedRequire (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623:559:15)
    at designatedRequire (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623:584:74)
    at requireRelative (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623:589:12)
    at require (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623:622:16)
    at (module ep_etherpad-lite/static/js/pad_utils.js) (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/javascripts/lib/ep_etherpad-lite/static/js/pad.js?callback=require.define&v=d3843623:1154:21)
    at loadModule (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623:410:11)
    at after (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623:434:11)
    at _find (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623:446:11)
    at _moduleAtPath (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623:452:5)
    at moduleAtPathSync (http://ui-541b39717c0c6c104482ee82abb2e860.local.sandstorm.io:6090/static/js/require-kernel.js?v=d3843623:488:7)
```

...in the browser. I checked, that module does exist under node_modules.

As part of this I removed what was patch 13, which just deleted some logging statements; there were conflicts and it didn't seem like the sort of thing that was worth having to actually maintain a patch over.

---

once this is actually working, i want to revisit the firefox workaround patch & the SameSite=Lax thing; those seem vaguely suspicious to me and the src->srcdoc change upstream might make stuff work?

Putting this down for the night, might try again later.

zenhack commented:
looking back at this again and it's not obvious to me what's going on re: the stack trace; it looks like there's some module resolution magic that's not working out correctly.

@orblivion does any of this look familiar to you, or is this a completely new problem with the new version?

orblivion commented:
A file that exists in node_modules but is not found by Etherpad does feel
vaguely familiar. Not seeing anything in the commit history from a very
quick look.

This isn't a vagrant problem is it? You did that weird bind mount
hackaround?
Message ID: <sandstormports/etherpad-lite-sandstorm/pull/6/c1358869190@
github.com>

>

zenhack commented:
Quoting Daniel Krol (2022-12-20 12:26:31)

>    This isn't a vagrant problem is it? You did that weird bind mount
>    hackaround?

This should be covered by build.sh iiuc? Or is there something I need to
do manually?

orblivion commented:
Yeah I guess it's automatic. It's the only thing I can think of.

zenhack committed:
First pass updating to 1.8.18

- Got the patches to apply cleanly.
- Dropped a patch that was disabling logging; had conflicts and doesn't
  seem worth the maintenance burden.

UI fails to display; instead we get the error:

```
An error occurred while loading the page:

    Error: The module at "js-cookie/dist/js.cookie" does not exist.

in http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36
at line 559:15

Stack trace:

    _designatedRequire@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:559:15
    designatedRequire@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:584:74
    requireRelative@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:589:12
    require@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:622:16
    (module ep_etherpad-lite/static/js/pad_utils.js)@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/javascripts/lib/ep_etherpad-lite/static/js/pad.js?callback=require.define&v=86a1ec36:1154:21
    loadModule@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:410:21
    after@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:434:21
    _find@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:446:11
    _moduleAtPath@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:452:10
    moduleAtPathSync@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:488:20
    _designatedRequire@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:556:18
    designatedRequire@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:584:74
    requireRelative@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:589:12
    require@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:622:16
    (module ep_etherpad-lite/static/js/pad.js)@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/javascripts/lib/ep_etherpad-lite/static/js/pad.js?callback=require.define&v=86a1ec36:33:24
    loadModule@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:410:21
    after@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:434:21
    _find@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:446:11
    _moduleAtPath@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:452:10
    moduleAtPathSync@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:488:20
    _designatedRequire@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:556:18
    designatedRequire@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:584:74
    requireRelative@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:589:12
    require@http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/static/js/require-kernel.js?v=86a1ec36:622:16
    @http://ui-a4aff542c13c061f703d85cf0396f6c6.local.sandstorm.io:6090/p/main:914:28
```

...in the browser. More investigation is needed.

zenhack commented:
Fixed the merge conflicts. Still broken though...

zenhack committed:
Update sandstorm-files.list

orblivion committed:
Install async-stacktrace explicitly, and also comment on node-pre-gyp

orblivion committed:
Etherpad repo fix for 1.8.14 -> 1.8.18 upgrade

Updated some patches related to one block of code in PadMessageHandler that
were not well in line with the upstream changes.

zenhack committed:
Bump version numbers

zenhack committed:
Remove stale part of comment.

This referred to node-pre-gyp, which has moved up above.

zenhack committed:
Update sandstorm-files.list

ocdtrekkie commented:
Looks reasonable to me. Did it magic away our CSP problem too, or do we still need to fix that?

zenhack commented:
No, that's still there.

ocdtrekkie reviewed (approved):
Reviewed commits and installed experimental package. Looks good to me.
